tox.ini

```diff
@@ -0,0 +1,29 @@
+[testenv]
+deps =
```

Comment:
Note-to-self: `tox.ini`'s `deps` directive supports `requirements.txt`-format files.

We may be able to update `testenv` and/or `lint` to read those from a shared location.

Reply:
Use cases I'm aware of:

1. GitHub Actions unit tests: run isolated tests using `tox` -- install package & dependencies from `pyproject.toml`, but without any other dependencies
2. Run unit tests locally: with or without package installed via `pyproject.toml` -- run tests using `unittest-parallel`, or `python -m unittest` directly; requires package dependencies installed, preferably into a virtualenv
3. Run GitHub Actions lint checks: run isolated linting using `tox` -- install lint tools
4. Run pre-commit checks locally: ideally using an isolated virtualenv (confirm: does pre-commit do this by default?), run `flake8`, `black`, etc

It's possible that items (3) and (4) can be consolidated.

Developers should be able to _choose_ to run (2) and (4) if they like locally - (1) and (3) will help them to identify problems if they don't (it's personal - and sometimes situational - preference about whether to be strict locally and/or use cloud compute to detect problems, especially bearing in mind multi-platform support)

At the moment as of this branch at 58ad0d2ec2ebfa5d9c83f77369cbe6e71fc1d6b4, the linting dependency list is duplicated between `tox.ini` (tox lint), `requirements-dev.txt`, and `.pre-commit-config.yaml` - all slightly divergent.

Could we remove that duplication?  Yep: linting/checking behaviours should be isolated.  And `pre-commit` seems designed for that purpose; so perhaps `tox` should run/install `pre-commit` to achieve lint checks (both in continuous integration and locally).

Regardless: `tox` itself seems like a unit testing improvement in terms of test isolation, so let's begin using it for continuous integration.